src/features/utils/event-buffer.js

```diff
@@ -44,18 +49,20 @@ export class EventBuffer {
   /**
    * Add feature-processed event to our buffer. If this event would cause our total raw size to exceed the set max payload size, it is dropped.
    * @param {any} event - any primitive type or object
+   * @param {boolean} [target] - target info if present
    * @param {number} [evaluatedSize] - the evalated size of the event, if already done so before storing in the event buffer
    * @returns {Boolean} true if successfully added; false otherwise
    */
-  add (event, evaluatedSize) {
+  add (event, target, evaluatedSize) {
     const addSize = evaluatedSize || stringify(event)?.length || 0 // (estimate) # of bytes a directly stringified event it would take to send
     if (this.#rawBytes + addSize > this.maxPayloadSize) {
       const smTag = inject => `EventBuffer/${inject}/Dropped/Bytes`
       this.featureAgg?.reportSupportabilityMetric(smTag(this.featureAgg.featureName), addSize) // bytes dropped for this feature will aggregate with this metric tag
       this.featureAgg?.reportSupportabilityMetric(smTag('Combined'), addSize) // all bytes dropped across all features will aggregate with this metric tag
       return false
     }
-    this.#buffer.push(event)
+    const hasV2Data = isValidMFETarget(target)
```

Comment:
Maybe consider adding an early return here if we already know target is undefined or null.  

Perhaps it'd be more performant if we can pre-determine the event's data version prior to calling `add` and cache that info somewhere so we don't have to check repeatedly 🤔  

```suggestion
    const hasV2Data = target ? isValidMFETarget(target) : false
```